chain/network-primitives/src/types.rs

```diff
@@ -332,12 +360,7 @@ pub enum NetworkViewClientResponses {
     /// Headers response.
     BlockHeaders(Vec<BlockHeader>),
     /// Chain information.
-    ChainInfo {
-        genesis_id: GenesisId,
-        height: BlockHeight,
-        tracked_shards: Vec<ShardId>,
-        archival: bool,
-    },
+    GetChainInfo(ChainInfo),
```

Comment:
There are NetworkViewClientMessages and NetworkViewClientResponses have matching names of variants so that it is obvious which response corresponds to each request. "Response" suffix is included in the name of the enum.
I can rename it from "GetChainInfo" to "ChainInfo" in both, if you want.

In fact for type safety, it would be better to have a separate message-response pair rather than pack everything into an enum, but that's what it is for now.

Reply:
obsolete